Author: Gregor Herrmann <gregoa@debian.org>
Subject: fix problems with dates in XML::Feed:Atom;
 Note according to changelog for 0.21 this was only partly adopted upstream
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=380498
Bug: http://rt.cpan.org/Ticket/Display.html?id=20763
Last-Update: 2011-07-17
Reviewed-by: Nicholas Bamber <nicholas@periapt.co.uk>
--- a/lib/XML/Feed/Format/Atom.pm
+++ b/lib/XML/Feed/Format/Atom.pm
@@ -268,7 +268,9 @@
     if (@_) {
         $entry->{entry}->issued(DateTime::Format::W3CDTF->format_datetime($_[0])) if $_[0];
     } else {
-        $entry->{entry}->issued ? iso2dt($entry->{entry}->issued) : undef;
+        return iso2dt($entry->{entry}->issued) if $entry->{entry}->issued;
+        return iso2dt($entry->{entry}->published) if $entry->{entry}->published;
+        return undef;
     }
 }
 
